backend_new/test/unit/services/permission.service.spec.ts

```diff
@@ -0,0 +1,344 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { newEnforcer } from 'casbin';
+import path from 'path';
+import { UserRoleEnum } from '../../../src/enums/permissions/user-role-enum';
+import { User } from '../../../src/dtos/users/user.dto';
+import { PermissionService } from '../../../src/services/permission.service';
+import { PrismaService } from '../../../src/services/prisma.service';
+import { permissionActions } from '../../../src/enums/permissions/permission-actions-enum';
+
+describe('Testing permission service', () => {
+  let service: PermissionService;
+  let prisma: PrismaService;
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [PermissionService, PrismaService],
+    }).compile();
+
+    service = module.get<PermissionService>(PermissionService);
+    prisma = module.get<PrismaService>(PrismaService);
+  });
+
+  it('should add admin user role for user', async () => {
+    const e = await newEnforcer(
+      path.join(
+        __dirname,
+        '../../../src/permission-configs',
+        'permission_model.conf',
+      ),
+      path.join(
+        __dirname,
+        '../../../src/permission-configs',
+        'permission_policy.csv',
+      ),
+    );
+
+    const user = {
+      id: 'example id',
+      userRoles: {
+        isAdmin: true,
```

Comment:
If this is already true, I'm not following how this is testing that addUserPermissions works

Reply:
ah this is a bit confusing since we have user roles internally and casbin has user roles too 

what this is basically saying is that if we have a user that has the isAdmin role in our DB then we we go to add user permissions casbin adds that they are an admin as well 

you can see that in backend_new/src/services/permission.service.ts ~ line 65